hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/ExpiryValidatorImpl.java

```diff
@@ -74,7 +74,7 @@ public ExpiryMeta resolveCreationAttempt(
         context.attributeValidator().validateExpiry(effectiveExpiry);
 
         // Even if the effective expiry is valid, we still also require any explicit auto-renew period to be valid
-        if (creationMeta.hasAutoRenewPeriod()) {
+        if (creationMeta.hasAutoRenewPeriod() && creationMeta.autoRenewPeriod() != 0) {
```

Comment:
Probably should be `>=`, so we're defensive against an accidental negative value.

Reply:
ExpiryMeta.autoRenewPeriod could have value Long.MIN_VALUE. Here is only skip 0 check.

Reply:
Interesting, we allow negative auto renew periods?  Seems like that might cause unexpected issues.  Something to note for later.